### PR TITLE
Improve the handling nullValue of @Compare #1389 

### PR DESCRIPTION
**Pull request by btyoshikawaa**
I reflected the changes of https://github.com/terasolunaorg/terasoluna-gfw/pull/416 to guideline.

This branch rebased on [issues/1389_change-attribute-name-of-compare](https://github.com/terasolunaorg/guideline/tree/issues/1389_change-attribute-name-of-compare), because related change `@Compare` attributes.
Please review #1389 to check cd1b68e1f8bbda7dc52169c5d21de05f37324db4 and 9b60f4deca1e3b67e3316137d2d0fd880caa903d.

### CODE REVIEW

File: source/ArchitectureInDetail/Validation.rst [hunk 1]
```diff
@@ -4002,18 +4015,79 @@ terasoluna-gfw-validatorのチェックルール
...
+                     right = "to",
                      operator = Compare.Operator.LESS_THAN_OR_EQUAL)
              public class Period {
                  private Date from;
                  private Date to;
              }
 
+.. note:: **相関項目チェックにおける入力必須について**
+  
+  単項目チェックにおいては、入力値が入力されている（\ ``null``\ でない）かどうかは \ ``@NotNull``\ を併用してチェックすればよい。しかし、相関項目チェックにおいては、「どちら一方でも入力した場合は、もう一方の入力を強制する」といった、 \ ``@NotNull``\ の併用だけでは実現できない場合がある。このため、\ ``@Compare``\ では、チェック対象の入力必須を制御する \ ``requireBoth``\ 属性を提供しており、これを併用して要件に応じたチェックを実装することができる。
```
Comment (ikeyat) [inline source/ArchitectureInDetail/Validation.rst]: - [ ] In my understanding, to enable ``requireBoth`` in Spring MVC, registration of ``StringTrimmerEditor`` is required. Please notice this fact and add link here (Note or Tips are useful for writing this).
    http://terasolunaorg.github.io/guideline/5.0.1.RELEASE/ja/ArchitectureInDetail/Validation.html#null

Comment (btyoshikawaa) [inline source/ArchitectureInDetail/Validation.rst]: @ikeyat 
Thanks for your comments.
I've reflected. Please show 9194c4a82a63e586948b32ae3463f8e9f2e9bd93.

**Comment by ikeyat** [comment]
@btyoshikawaa 
Check my comment in filechanges and fix conflicts

**Comment by ikeyat** [comment]
@btyoshikawaa Please fix conflicts. Sorry for my late response that may caused conflicts.

**Comment by btyoshikawaa** [comment]
@ikeyat 
I've rebased and fixed all conflicts. You should be able to merge to the master.

**Pull request merged**